### PR TITLE
Cities Skylines Multiplayer API Implementation

### PR DESCRIPTION
**Pull request by MightyWizard**
Implemented the new Cities Skylines Multiplayer API into TMPE using the notifier.
https://github.com/CitiesSkylinesMultiplayer/CSM/pull/252
Cities Skylines Multiplayer API is currently in review but should be pulled into master soon.
Feel free to ask question and make any changes 

Features that are not synced are 
- Time traffic lights
- No speed limit
- Removal of vehicles ( single or all)

### CODE REVIEW

**Comment by MightyWizard** [comment]
There could potentially by an issue with ignores not being set when notifications are received in csmConnections, so data from TMPENotificationHandler.cs is seen as changes and repeatedly sent

**Review by krzychu124 — CHANGES_REQUESTED** [review]
TL;DR
Probably 95% of the logic you proposed here shouldn't be a part of TM:PE and remaining 5% we can easily expose through our API in form of events.

---
Main issue: PR introduces "hard dependency" to protobuf (and its inner dependencies) and CSM API. From observation and frequency of requests for implementing support for CSM in the past I can say that 95% of TMPE users don't use that mod. 

Another problem with "hard dependency" is if I copy CSM Api dll to workshop folder it will cause issues if it gets updated by CSM updates and ship later (runtime versioning nightmare). On the other hand if I don't copy CSM API dll and there is no CSM available then TMPE won't load properly, additionally causing other assembly loading issues.

Protobuf requires multiple other network related libraries to load correctly (not referenced anywhere here and C:S is known of assembly resolve issues. Without dependencies it will cause `ReflectionTypeLoadException` - probably the worst and most annoying runtime exception in C:S and once thrown upon game start it will basically break 95% of enabled mods from the workshop (old story of NExt2 compatibility issue combined with Harmony, thankfully resolved somehow).

Personally I can't even run CSM alone on clean game install, it spams the log with `ReflectionTypeLoadException` so adding more mods is pointless since they'll probably won't load/work properly because that exception. Not to mention that CSM shouldn't be shipped with own version of `Harmony.dll` if combined with mods requiring `CitiesHarmony.API` (e.g. TM:PE)

### My suggestion
Create a "bridge/adapter" assembly which could test for TMPE availability first and combine TM:PE API and CSM API then handle communication and client-server synchronization. I will be TM:PE version independent and quite flexible. You will ship that adapter with CSM mod and do similar thing for other mods if required. 

Anyways, if you really plan to support more mods, it will be easier to use Harmony and add postfixes/prefixes (to mimic notifications from a mod) rather than trying to introduce changes to a mod you are going to provide support to.

**Comment by kianzarrin** [comment]
I don't recall closing this. this probably got closed because I merged my notifier changes.
